analysis/src/npm.js

```diff
@@ -60,18 +58,17 @@ class NPM {
   * @return {Promise} A promise handling the prune operation.
   */
   prune() {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
```

Comment:
just make this whole function async/await instead of making a promise yourself?

Reply:
Not sure how that works with the reject params, sorry.